### DISCUSSION TITLE
Subscription status, other transaction id (Only AppStore)

### DISCUSSION THREAD

**Discussion by ghost** [imdhemy/laravel-in-app-purchases#393] [Q&A]
I need to know two things, first, is there any way to get the subscription status from AppStore events?.. second, is coming from AppStore another id like a transaction id that be different from the unique id, cause i need to log every event for billing reasons and i need to associate every event with an id?

**Comment by imdhemy** [comment]
Hi @vincy-dev93 
In the documentation, you can find an example of [getting the subscription data](https://imdhemy.com/laravel-iap-docs/docs/get-started/event-listeners#handling-an-event-example). 

```php
 $subscription = $event->getServerNotification()->getSubscription();
```

For the other question, maybe the `getNotificationUUID()` could help.

https://github.com/imdhemy/appstore-iap/blob/91bf598a468c453797dbca10c87bf53c7f4956ca/src/ServerNotifications/V2DecodedPayload.php#L120-L128

**Reply by ghost** [reply]
Sorry again @imdhemy but , I can't find how can i have this statuses apple says:

1
The auto-renewable subscription is active.

2
The auto-renewable subscription is expired.

3
The auto-renewable subscription is in a billing retry period.

4
The auto-renewable subscription is in a Billing Grace Period.

5
The auto-renewable subscription is revoked.

Any suggestion?

**Reply by imdhemy** [reply]
Within the event handler you can know the status from the event name. Or from the notification you use the `getType()` method. It will return a string containing one of the following types:

https://github.com/imdhemy/appstore-iap/blob/91bf598a468c453797dbca10c87bf53c7f4956ca/src/ServerNotifications/V2DecodedPayload.php#L22-L36

But if you need the status as an integer, you need to send  a `verifyReceipt()` request. 

I guess the event name or the type is enough for any use-case.

**Reply by ghost** [reply]
Thanks @imdhemy, I really appreciate your answer, I'm already using the getType() and getSubType() methods, both are very helpful.

**Reply by ghost** [reply]
Finally I've used this transactionId to identify them...

`$subscriptionContract = $event->getServerNotification()->getSubscription();`
`$v2DecodedPayload = $subscriptionContract->getProviderRepresentation();`
`$transactionId = $v2DecodedPayload->getTransactionInfo()->getTransactionId();`

This is a different id for every purchased or renewal events, perfect for billing logs
